splink/linker.py

```diff
@@ -378,6 +378,43 @@ def _execute_sql_against_backend(self, sql, templated_name, physical_name):
             f"_execute_sql_against_backend not implemented for {type(self)}"
         )
 
+    def register_table(self, input, table_name):
+        """
+        Register a table to your backend database, to be used in one of the
+        splink methods, or simply to allow querying.
+
+        Tables can be of type: dictionary, record level dictionary,
+        pandas dataframe, pyarrow table and in the spark case, a spark df.
+
+        Examples:
+            >>> test_dict = {"a": [666,777,888],"b": [4,5,6]}
+            >>> linker.register_table(test_dict, "test_dict")
+            >>> linker.query_sql("select * from test_dict")
+
+        Args:
+            input: The data you wish to register. This can be either a dictionary,
+                pandas dataframe, pyarrow table or a spark dataframe.
+            table_name (str): The name you wish to assign to the table.
+        """
+
+        raise NotImplementedError(f"register_table not implemented for {type(self)}")
+
+    def query_sql(self, sql):
```

Comment:
DO we definitely want this to return a pandas dataframe?  

I'm not sure that's such a good idea with big data so I think probably it should return a `SplinkDataframe`, which would also be more consistent with the rest of the API.

(Sorry if im missing something obvious)